<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-1-g15f4949  Build Version: 1.6.7594.29634
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="lewansoul_xarm"
  xmlns:xacro="http://www.ros.org/wiki/xacro" >
  <xacro:property name="camera_link" value="0.1" /> <!-- Size of square 'camera' box -->
  
  <gazebo>
   <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
     <robotNamespace>/rrbot</robotNamespace>
   </plugin>
  
 </gazebo>

  <gazebo reference="camera_link">
  
    <sensor type="camera" name="camera1">
      <update_rate>30.0</update_rate>
      
      <camera name="head">
        <horizontal_fov>1.3962634</horizontal_fov>
        <image>
          <width>800</width>
          <height>800</height>
          <format>R8G8B8</format>
        </image>
        <clip>
          <near>0.02</near>
          <far>300</far>
        </clip>
        <noise>
          <type>gaussian</type>
          <!-- Noise is sampled independently per pixel on each frame.
               That pixel's noise value is added to each of its color
               channels, which at that point lie in the range [0,1]. -->
          <mean>0.0</mean>
          <stddev>0.007</stddev>
        </noise>
      </camera>
      
      <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
        <alwaysOn>true</alwaysOn>
        <updateRate>0.0</updateRate>
        <cameraName>rrbot/camera1</cameraName>
        <imageTopicName>image_raw</imageTopicName>
        <cameraInfoTopicName>camera_info</cameraInfoTopicName>
        <frameName>camera_link</frameName>
        <hackBaseline>0.07</hackBaseline>
        <distortionK1>0.0</distortionK1>
        <distortionK2>0.0</distortionK2>
        <distortionK3>0.0</distortionK3>
        <distortionT1>0.0</distortionT1>
        <distortionT2>0.0</distortionT2>
      </plugin>
      
    </sensor>
    
  </gazebo>
 
 
 <!-- ___________________________________________________________LINK and JOINT ____________________________________________________ -->
  
 <link
    name="base_link">
    <inertial>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <!--origin
        xyz="5.62597317247015E-06 0.0211535754066861 -0.000552905012225837"
        rpy="0 0 0" /-->
      <mass
        value="1.1436112237347" />
      <inertia
        ixx="0.000159180717934752"
        ixy="2.83202436887395E-22"
        ixz="1.13075425821565E-21"
        iyy="0.000295963658183954"
        iyz="4.3547292877157E-08"
        izz="0.000141019350389223" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://lewansoul_xarm/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://lewansoul_xarm/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  
  
  <link
    name="link_1">
    <inertial>
      <origin
        xyz="-0.00573780856637832 0.0246764862397383 -0.00276539232704796"
        rpy="0 0 0" />
      <mass
        value="0.0479211105385059" />
      <inertia
        ixx="1.08121838088991E-05"
        ixy="3.16914341671095E-13"
        ixz="-1.15653005267957E-06"
        iyy="2.1109129120512E-05"
        iyz="3.16914341369538E-13"
        izz="1.26654818978706E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://lewansoul_xarm/meshes/link_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.250980392156863 0.250980392156863 0.250980392156863 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://lewansoul_xarm/meshes/link_1.STL" />
      </geometry>
    </collision>
  </link>
  
  <joint
    name="joint_1"
    type="revolute">
    <origin
      xyz="0 0 0.02958"
      rpy="1.5708 0 -1.6316" />
    <parent
      link="base_link" />
    <child
      link="link_1" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-6.28319"
      upper="6.28319"
      effort="10"
      velocity="0.01" />
  </joint>
  
  <link
    name="link_2">
    <inertial>
      <origin
        xyz="0.0172168172964557 -0.0156153859453483 -0.0432893744013595"
        rpy="0 0 0" />
      <mass
        value="0.0270439753276969" />
      <inertia
        ixx="9.22319968691187E-06"
        ixy="3.94332228926969E-13"
        ixz="9.69593263039659E-07"
        iyy="5.92009760396816E-06"
        iyz="-2.13005855040871E-13"
        izz="7.17090699107036E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://lewansoul_xarm/meshes/link_2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.250980392156863 0.250980392156863 0.250980392156863 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://lewansoul_xarm/meshes/link_2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_2"
    type="revolute">
    <origin
      xyz="0.0085927 0.035475 -0.012421"
      rpy="-1.7405 0.4322 2.7533" />
    <parent
      link="link_1" />
    <child
      link="link_2" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="10"
      velocity="0.01" />
  </joint>
  <link
    name="link_3">
    <inertial>
      <origin
        xyz="0.0448309767527862 -0.0164587912318094 0.0164117731401168"
        rpy="0 0 0" />
      <mass
        value="0.0638914701308421" />
      <inertia
        ixx="7.0703356990677E-06"
        ixy="3.94337815638652E-13"
        ixz="-2.04127157420929E-06"
        iyy="1.16829223609899E-05"
        iyz="-2.13021322603642E-13"
        izz="1.1899049169965E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://lewansoul_xarm/meshes/link_3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.250980392156863 0.250980392156863 0.250980392156863 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://lewansoul_xarm/meshes/link_3.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_3"
    type="revolute">
    <origin
      xyz="0.036073 0 -0.0907"
      rpy="3.1416 1.5538 3.1416" />
    <parent
      link="link_2" />
    <child
      link="link_3" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="10"
      velocity="0.01" />
  </joint>
  <link
    name="link_4">
    <inertial>
      <origin
        xyz="-0.0295970834275255 -0.0128772765655342 -0.017591135236586"
        rpy="0 0 0" />
      <mass
        value="0.0363295532641635" />
      <inertia
        ixx="9.47176390528689E-06"
        ixy="3.94331980111041E-13"
        ixz="-2.51294097066946E-07"
        iyy="9.85285609564443E-06"
        iyz="-2.13005931379715E-13"
        izz="8.32887226666747E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://lewansoul_xarm/meshes/link_4.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.250980392156863 0.250980392156863 0.250980392156863 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://lewansoul_xarm/meshes/link_4.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_4"
    type="revolute">
    <origin
      xyz="0.092271 -0.00131 0.033779"
      rpy="-3.1416 0 3.1416" />
    <parent
      link="link_3" />
    <child
      link="link_4" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="10"
      velocity="0.01" />
  </joint>
  <!--link
    name="link_5">
    <inertial>
      <origin
        xyz="0.00956645686615627 0.034288981355211 -0.00283882421708866"
        rpy="0 0 0" />
      <mass
        value="0.0489913982778047" />
      <inertia
        ixx="9.60443354651095E-06"
        ixy="8.20842431174883E-11"
        ixz="1.12113133649776E-07"
        iyy="6.43951133454609E-06"
        iyz="-2.39894432330848E-09"
        izz="4.38405850313509E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://lewansoul_xarm/meshes/link_5.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.250980392156863 0.250980392156863 0.250980392156863 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://lewansoul_xarm/meshes/link_5.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_5"
    type="revolute">
    <origin
      xyz="-0.049476 -0.012475 -0.012475"
      rpy="-1.7864 -1.2936 -2.9177" />
    <parent
      link="link_4" />
    <child
      link="link_5" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="10"
      velocity="0.01" />
  <joint-->
  
    <!-- _____________________________________Camera ______________________________________-->
  <link name="camera_link">
    <collision>
      <origin xyz="0 0.1 0.15" rpy="0 0 0"/>
      <geometry>
    <box size="0.01 0.01 0.01"/>
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0.1 0.15" rpy="0 0 0"/>
      <geometry>
    <box size="0.01 0.01 0.01"/>
      </geometry>
      <material name=""/>
      <color
          rgba="0.250980392156863 0.250980392156863 0.250980392156863 1" />
    </visual>
  </link>
	
 <joint name="camera_joint" type="fixed">
    <axis xyz="0 1 0.1" />
    <origin xyz="0 0.1 0.1 " rpy="0 0 0"/>
    <parent link="base_link"/>
    <child link="camera_link"/>
  </joint>
  
  <link name="world"/>
  
   <joint name="worldfix" type="fixed">
    <axis xyz="0 0 0" />
    <origin xyz="0 0 0 " rpy="0 0 0"/>
    <parent link="world"/>
    <child link="base_link"/>
  </joint>



<!--_________________________________________TRANSMITTION________________________________________-->

 <transmission name="tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_1">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="tran2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_2">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor2">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  
 <transmission name="tran3">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_3">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor3">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="tran4">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_4">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor4">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  
   <!--transmission name="tran5">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_5">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor5">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission-->


  
</robot>


